providers/os/detector/detector_win.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package detector
+
+import (
+	"strconv"
+
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/v11/providers/os/connection/shared"
+	win "go.mondoo.com/cnquery/v11/providers/os/detector/windows"
+	"go.mondoo.com/cnquery/v11/providers/os/registry"
+)
+
+func runtimeWindowsDetector(pf *inventory.Platform, conn shared.Connection) (bool, error) {
+	data, err := win.GetWmiInformation(conn)
+	if err != nil {
+		log.Debug().Err(err).Msg("could not gather wmi information")
+	}
+
+	pf.Name = "windows"
+	pf.Title = data.Caption
+
+	// instead of using windows major.minor.build.ubr we just use build.ubr since
+	// major and minor can be derived from the build version
+	pf.Version = data.BuildNumber
+
+	// FIXME: we need to ask wmic cpu get architecture
+	pf.Arch = data.OSArchitecture
+
+	if pf.Labels == nil {
+		pf.Labels = map[string]string{}
+	}
+	pf.Labels["windows.mondoo.com/product-type"] = data.ProductType
+
+	// optional: try to get the ubr number (win 10 + 2019)
+	current, err := win.GetWindowsOSBuild(conn)
+	if err == nil && current.UBR > 0 {
+		pf.Build = strconv.Itoa(current.UBR)
+	} else {
+		log.Debug().Err(err).Msg("could not parse windows current version")
```

Comment:
if `err` is `nil` but `urrent.UBR` is empty. We will panic here.

Reply:
yep, fixed